python/tests/test_combinatorics.py

```diff
@@ -481,6 +481,31 @@ def test_is_symmetrical(self):
         six_leaf_sym = RankTree(children=[three_leaf_asym, three_leaf_asym])
         assert six_leaf_sym.is_symmetrical()
 
+    @pytest.mark.parametrize("n", range(7))
+    def test_leaves(self, n):
+        for tree in RankTree.all_unlabelled_trees(n):
+            first_labelled_tree = RankTree.unrank(n, (tree.shape_rank(), 0))
+            assert list(first_labelled_tree.leaves()) == list(range(n))
+
+    @pytest.mark.parametrize("n", range(5))
+    def test_no_leaves(self, n):
+        for tree in RankTree.all_unlabelled_trees(n):
```

Comment:
This is excessive - testing this on one or two examples would be the same as iterating over all trees here.